Userland/Utilities/shot.cpp

```diff
@@ -5,28 +5,100 @@
 #include <LibCore/ArgsParser.h>
 #include <LibCore/DateTime.h>
 #include <LibCore/File.h>
 #include <LibGUI/Application.h>
 #include <LibGUI/Clipboard.h>
+#include <LibGUI/Painter.h>
+#include <LibGUI/Widget.h>
+#include <LibGUI/Window.h>
 #include <LibGUI/WindowServerConnection.h>
 #include <LibGfx/PNGWriter.h>
 #include <unistd.h>
 
+class SelectableLayover final : public GUI::Widget {
+    C_OBJECT(SelectableLayover)
+public:
+    SelectableLayover(GUI::Window* window)
+        : m_window(window) {};
+    virtual ~SelectableLayover() override {};
+    Optional<Gfx::IntRect> region() const
+    {
+        return m_region;
+    }
+
+private:
+    virtual void mousedown_event(GUI::MouseEvent& event) override
+    {
+        if (event.button() != GUI::MouseButton::Left)
+            return;
+
+        if (m_achor_point.has_value())
+            m_achor_point.clear();
+        else
+            m_achor_point = event.position();
+    };
+
+    virtual void mousemove_event(GUI::MouseEvent& event) override
+    {
+        m_last_point = event.position();
+        if (m_achor_point.has_value())
+            update();
+    };
+
+    virtual void paint_event(GUI::PaintEvent&) override
+    {
+        m_region = build_rect(m_achor_point, m_last_point);
+        GUI::Painter painter(*this);
+        painter.draw_rect(m_region.value_or(Gfx::IntRect()), Gfx::Color::Black);
+    }
+
+    virtual void keydown_event(GUI::KeyEvent& event) override
+    {
+        if (event.key() != Key_Return)
+            return;
+
+        m_window->close();
+    }
+
+    static Optional<Gfx::IntRect> build_rect(Optional<Gfx::IntPoint> const& p1, Optional<Gfx::IntPoint> const& p2)
+    {
+        if (!p1.has_value() || !p1.has_value())
+            return {};
+
+        Gfx::IntPoint point(
+            min(p1.value().x(), p2.value().x()),
+            min(p1.value().y(), p2.value().y()));
+        Gfx::Size size(
+            abs(p1.value().x() - p2.value().x()),
+            abs(p1.value().y() - p2.value().y()));
+
+        return Gfx::IntRect(point, size);
+    }
+
+    Optional<Gfx::IntPoint> m_last_point {};
+    Optional<Gfx::IntPoint> m_achor_point {};
+    Optional<Gfx::IntRect> m_region {};
+    GUI::Window* m_window;
+};
+
 int main(int argc, char** argv)
 {
     Core::ArgsParser args_parser;
 
     String output_path;
     bool output_to_clipboard = false;
     int delay = 0;
+    bool select_region = false;
     int screen = -1;
 
     args_parser.add_positional_argument(output_path, "Output filename", "output", Core::ArgsParser::Required::No);
     args_parser.add_option(output_to_clipboard, "Output to clipboard", "clipboard", 'c');
     args_parser.add_option(delay, "Seconds to wait before taking a screenshot", "delay", 'd', "seconds");
     args_parser.add_option(screen, "The index of the screen (default: -1 for all screens)", "screen", 's', "index");
+    args_parser.add_option(select_region, "Select a region to capture", "region", 'r');
```

Comment:
It'd also be neat to make this scriptable, so you can say `shot --rect 10,10,40,40` or something to that effect... though that can be done in a later PR/change too.

Reply:
Good idea! I will look into that in a later PR. Which one makes more sense to define the rect x,y,w,h or x1,y1,x2,y2?

Reply:
I personally like x,y,w,h but we can bikeshed that when needed :P